docs/wasmer-js/node-modules/examples/hello-world/hello-world.md

```diff
@@ -6,113 +6,166 @@ sidebar_label: Hello World
 
-This example will be run in the browser, bundled and served by [Parcel](https://parceljs.org/). 
+`JavaScript` --> `WebAssembly` --> `Some native "OS" function`
 
-However, `@wasmer/wasi` and `@wasmer/wasmfs` also work in Node, and the code examples from this guide can be used as a loose example, as long as the ES6 syntax is replaced with what would be the node equivalent.
+`WASI` is the bridge that allows a WebAssembly module to invoke a native "OS" function.
 
-First, let's set up a small JavaScript project using Parcel. To do this, we will create a new directory, and initialize it as an node project with npm:
+> ### As an Aside...  
+> The term "OS" is in double quotes to indicate that the native function being called might not actually belong to the underlying operating system.  
+> In reality, this function belongs to the host environment within which this WebAssembly module is running, and in this particular case, this is the environment provided by the browser, not the underlying operating system.  
+> However, from a WebAssembly point of view, we don't need to care about this detail. All we need to know is that this function exists, and we can call it (as long as we're careful)!
+
+
+In this case, we want to run the simple WASM module [as-echo](https://github.com/torch2424/as-echo); however, we know this module writes its output to standard out, which in turn, requires access to the functionality found in the underlying "OS".  This means that unless we provide the following two `WASI` packages, this module cannot run:
+
+| Package Name | Description
+|---|---|
+| `@wasmer/wasi` | Bridges the gap between the black-box world of a WebAssembly module and functionality available from the host environment
+| `@wasmer/wasmfs` | A sandboxed filesystem with which `@wasmer/wasi` can interact
+
+The `as-echo` module simply receives a text string (in our case `"Hello World!"`) and echoes it back via standard output `/dev/stdout`.
+
+This example will be bundled and served by [Parcel](https://parceljs.org/) and run in the browser.
+
+The `@wasmer/wasi` and `@wasmer/wasmfs` also work in NodeJS, and the code examples from this guide can be used as a loose example &mdash; as long as the ES6 syntax is replaced with the equivalent NodeJS coding.
+
+## Prerequisites
+
+Make sure [Parcel](https://parceljs.org/) has been installed and is available from the command line
 
 ```bash
-mkdir my-wasmer-js-app
-cd my-wasmer-js-app
-npm init
+$ npm install -g parcel
 ```
 
-Then, answer the prompts from `npm init`. After this, we will install the `parcel-bundler` package, as well as `parcel-plugin-static-files-copy` to allow serving our wasm files as a static asset:
+> ### Mac users
+> Before the installation of Parcel will work on a Mac, you must first install the [Xcode Command Line Tools](https://developer.apple.com/download/more/?=for%20Xcode)
 
-`npm install --save-dev parcel-bundler parcel-plugin-static-files-copy`
+## Run Example from Git Clone
```

Comment:
Well OK, but if any assumptions being made, they should be clearly stated.

I'll reinstate the step-by-step instructions rather than recommending to clone the repo...

Reply:
@ChrisWhealy Thank you! 😄